airflow/contrib/operators/sagemaker_endpoint_operator.py

```diff
@@ -0,0 +1,149 @@
+    :type wait_for_completion: bool
+    :param check_interval: If wait is set to True, this is the time interval, in seconds, that this operation waits
+        before polling the status of the endpoint creation.
+    :type check_interval: int
+    :param max_ingestion_time: If wait is set to True, this operation fails if the endpoint creation doesn't finish
+        within max_ingestion_time seconds. If you set this parameter to None it never times out.
+    :type max_ingestion_time: int
+    :param operation: Whether to create an endpoint or update an endpoint. Must be either 'create or 'update'.
+    :type operation: str
+    """  # noqa
+
+    @apply_defaults
+    def __init__(self,
+                 config,
+                 wait_for_completion=True,
+                 check_interval=30,
+                 max_ingestion_time=None,
+                 operation='create',
+                 *args, **kwargs):
+        super(SageMakerEndpointOperator, self).__init__(config=config,
+                                                        *args, **kwargs)
+
+        self.config = config
+        self.wait_for_completion = wait_for_completion
+        self.check_interval = check_interval
+        self.max_ingestion_time = max_ingestion_time
+        self.operation = operation.lower()
+        self.create_integer_fields()
+
+    def create_integer_fields(self):
+        if 'EndpointConfig' in self.config:
+            self.integer_fields = [
+                ['EndpointConfig', 'ProductionVariants', 'InitialInstanceCount']
+            ]
+
+    def expand_role(self):
+        if 'Model' not in self.config:
+            return
+        hook = AwsHook(self.aws_conn_id)
+        config = self.config['Model']
+        if 'ExecutionRoleArn' in config:
+            config['ExecutionRoleArn'] = hook.expand_role(config['ExecutionRoleArn'])
+
+    def execute(self, context):
+        self.preprocess_config()
+
+        model_info = self.config['Model'] if 'Model' in self.config else None
+        endpoint_config_info = self.config['EndpointConfig'] if 'EndpointConfig' in self.config else None
+        endpoint_info = self.config['Endpoint'] if 'Endpoint' in self.config else self.config
+
+        if model_info:
+            self.log.info('Creating SageMaker model %s.', model_info['ModelName'])
+            self.hook.create_model(model_info)
+
+        if endpoint_config_info:
+            self.log.info('Creating endpoint config %s.', endpoint_config_info['EndpointConfigName'])
+            self.hook.create_endpoint_config(endpoint_config_info)
+
+        if self.operation == 'create':
+            sagemaker_operation = self.hook.create_endpoint
+            log_str = 'Creating'
+        elif self.operation == 'update':
+            sagemaker_operation = self.hook.update_endpoint
+            log_str = 'Updating'
+        else:
+            raise AirflowException('Invalid value! Argument operation has to be one of "create" and "update"')
```

Comment:
Could we validate this field in the constructor instead of execute time?

(Or perhaps both?)